runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangePlugin.java

```diff
@@ -441,11 +517,138 @@ public void onWidgetLoaded(WidgetLoaded event)
 	public void onScriptPostFired(ScriptPostFired event)
 	{
 		// GE offers setup init
-		if (event.getScriptId() != ScriptID.GE_OFFERS_SETUP_BUILD)
+		if (event.getScriptId() == ScriptID.GE_OFFERS_SETUP_BUILD)
+		{
+			rebuildGeText();
+		}
+		else if (event.getScriptId() == ScriptID.GE_ITEM_SEARCH && config.highlightSearchMatch())
+		{
+			highlightSearchMatches();
+		}
+	}
+
+	private void highlightSearchMatches()
+	{
+		String input = client.getVar(VarClientStr.INPUT_TEXT);
+
+		String highlightColorHex = String.format("%06x", config.highlightMatchColor().getRGB() & 0xFFFFFF);
+		String underlineTag = "<u=" + highlightColorHex + ">";
+
+		Widget results = client.getWidget(WidgetInfo.CHATBOX_GE_SEARCH_RESULTS);
+		Widget[] children = results.getDynamicChildren();
+		int resultCount = children.length / 3;
+
+		for (int i = 0; i < resultCount; i++)
+		{
+			Widget itemNameWidget = children[i * 3 + 1];
+			String itemName = itemNameWidget.getText();
+
+			List<Integer> indices;
+			if (wasFuzzySearch)
+			{
+				FuzzyScore fuzzy = new FuzzyScore(Locale.ENGLISH);
+				String otherName = itemName.replace('-', ' ');
+				if (!itemName.contains("-") || fuzzy.fuzzyScore(itemName, input) >= fuzzy.fuzzyScore(otherName, input))
+				{
+					indices = findFuzzyIndices(itemName, input);
+				}
+				else
+				{
+					indices = findFuzzyIndices(otherName, input);
+				}
+			}
+			else
+			{
+				indices = findSubstringIndices(itemName, input);
+			}
+			Collections.reverse(indices);
+
+			StringBuilder newItemName = new StringBuilder(itemName);
+			for (int index : indices)
+			{
+				if (wasFuzzySearch && (itemName.charAt(index) == ' ' || itemName.charAt(index) == '-'))
+				{
+					continue;
+				}
+				newItemName.insert(index + 1, "</u>");
+				newItemName.insert(index, underlineTag);
+			}
+
+			itemNameWidget.setText(newItemName.toString());
+		}
+	}
+
+	@Subscribe
+	public void onGrandExchangeSearched(GrandExchangeSearched event)
+	{
+		wasFuzzySearch = false;
+
+		GrandExchangeSearchMode searchMode = config.geSearchMode();
+		final String input = client.getVar(VarClientStr.INPUT_TEXT);
+		if (searchMode == GrandExchangeSearchMode.DEFAULT || input.isEmpty())
 		{
 			return;
 		}
-		rebuildGeText();
+
+		event.consume();
+
+		int resultCount = 0;
+		if (searchMode == GrandExchangeSearchMode.FUZZY_FALLBACK)
+		{
+			List<Integer> ids = IntStream.range(0, client.getItemCount())
+					.mapToObj(itemManager::getItemComposition)
+					.filter(item -> item.isTradeable() && item.getNote() == -1
+							&& item.getName().toLowerCase().contains(input))
```

Comment:
One indent less here
```suggestion
						&& item.getName().toLowerCase().contains(input))
```